ExtremeValueProject/DegenerateCDF.lean

```diff
@@ -24,24 +24,107 @@ lemma isDegenerate_def (F : CumulativeDistributionFunction) :
 /-- A c.d.f. F is degenerate if and only if it jumps from 0 to 1 at some point x₀. -/
 lemma isDegenerate_iff (F : CumulativeDistributionFunction) :
     F.IsDegenerate ↔ ∃ x₀, F.toFun = (Set.Ici x₀).indicator (fun _ ↦ 1) := by
-  sorry -- **Issue #12**
+  constructor
+  · intro is_degen
+
+    have obs (x : ℝ) : F x = 1 ↔ F x ≠ 0 := by
+      constructor
+      · intro h
+        exact ne_zero_of_eq_one h
+      · intro hx
+        cases is_degen x
+        · contradiction
+        · assumption
+
+    have approaches_zero := isGLB_of_tendsto_atBot F.mono F.tendsto_atBot
+    have reaches_zero : ∃ x : ℝ, F x = 0 := by
+      rw [← not_forall_not]
+      intro h
+      have F_one : F.toFun = fun (x : ℝ) => 1 := by
+        funext x
+        rw [obs]
+        exact h x
+      rw [F_one, range_const] at approaches_zero
+      have := IsGLB.unique approaches_zero isGLB_singleton
+      norm_num at this
+
+    have approaches_one := isLUB_of_tendsto_atTop F.mono F.tendsto_atTop
+    have reaches_one : ∃ x : ℝ, F x = 1 := by
+      rw [← not_forall_not]
+      intro h
+      have F_zero : F.toFun = fun (x : ℝ) => 0 := by
+        funext x
+        simp only [obs, not_not] at h
+        exact h x
+      rw [F_zero, range_const] at approaches_one
+      have := IsLUB.unique approaches_one isLUB_singleton
+      norm_num at this
+
+    have bounded_below : BddBelow {x : ℝ | F x = 1} := by
+      unfold BddBelow
+      obtain ⟨x₀, h⟩ := reaches_zero
+      use x₀
+      intro a (ha : F a = 1)
+      apply le_of_lt
+      apply Monotone.reflect_lt F.mono
+      rw [h, ha]
+      norm_num
+
+    let x₀ := sInf {x : ℝ | F x = 1}
+    have one_after_x₀ : ∀ x > x₀, F x = 1 := by
+      intro x hx
+      apply le_antisymm
+      · exact apply_le_one F x
+      · obtain ⟨x₁, ⟨is_one : F x₁ = 1, lt_x⟩⟩ := exists_lt_of_csInf_lt reaches_one hx
+        rw [← is_one]
+        exact F.mono (le_of_lt lt_x)
```

Comment:
Again, the following is your original proof just combined to fewer lines.

(To be clear, fewer lines is not always better, but here I think your proof is organized into very nice self-explanatory and intuitive `have` statements which are given very descriptive names, and for that reason within the have statements it is ok to golf a bit aggressively.)

```suggestion
      apply le_antisymm (apply_le_one F x)
      obtain ⟨x₁, ⟨is_one : F x₁ = 1, lt_x⟩⟩ := exists_lt_of_csInf_lt reaches_one hx
      simpa [← is_one] using F.mono (le_of_lt lt_x)
```